app/Controllers/Http/ExplorerController.js

```diff
@@ -1099,15 +1103,15 @@ class ExplorerController {
         .where('name', wsPath)
         .update({ settings: configStr })
     } else {
-      if (fs.existsSync(path)) {
-        fs.writeFileSync(path, configStr, {encoding: 'utf8'});
+      if (await exists(path)) {
+        await writeFile(path, configStr, {encoding: 'utf8'});
         return true;
       }
     }
     return false;
   }
 
-  isDirectory = source => lstatSync(source).isDirectory();
+  isDirectory = async (source) => lstat(source).then(stat => stat.isDirectory());
```

Comment:
Right now the type of `isDirectry` is a promise. If you put an `await` before `async`, the variable is of type bool already and no one has to do an await on the isDirectory flag to read it.

Reply:
I agree that's a redundant `async`

Reply:
Forget what I said. I thought `isDirectory` is a local variable. But it is a short hand syntax for a member function. Then we should not add the `await` here to make `isDirectory` a bool, but it should stay a function. Whether you remove the `redundant` async or not, I don't care. From my point of view, this PR can be merged now :)